core/src/main/java/org/apache/accumulo/core/zookeeper/ZcNode.java

```diff
@@ -29,27 +29,42 @@
  * and children are obtained from zookeeper at different times. This class is structured so that
  * data can be obtained first and then children added later or visa veras.
  *
- * <p>
- * Four distinct states can be cached for a zookeeper node.
- * <ul>
- * <li>Can cache that a node does not exist in zookeeper. This state is represented by data, state,
- * and children all being null.</li>
- * <li>Can cache only the data for a zookeeper node. For this state data and stat are non-null while
- * children is null. Calling getChildren on node in this state will throw an exception.</li>
- * <li>Can cache only the children for a zookeeper node. For this state children is non-null while
- * data and stat are null. Calling getData or getStat on node in this state will throw an
- * exception.</li>
```

Comment:
Could keep the comments regarding what method calls in what state will result in an exception

Reply:
I updated the javadoc for the methods to be a more specific.